### PR TITLE
Current world orb

### PR DESCRIPTION
**Pull request by devOSRS**
Very rough and is a work-in progress, but made a simple overlay to show your current world over the world map orb.

Let me know what you think! https://github.com/runelite/runelite/issues/5920

screen shot 2018-10-12 at 1 00 31 am

### CODE REVIEW

**Comment by deathbeam** [comment]
you should probably use default font, this looks too obnoxious

**Comment by devOSRS** [comment]
default font was too small and when made bigger it was blurry, but I'll see what I can do

**Comment by deathbeam** [comment]
The default font would be the same size as the font next to rest of orbs right? I dont think that is too small.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java [hunk 1]
```diff
@@ -191,6 +198,8 @@ protected void startUp() throws Exception
 		{
 			clientToolbar.addNavigation(navButton);
 		}
+
+		overlayManager.add(worldHopperOverlay);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java]: This should be removed in `shutdown`

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		setPriority(OverlayPriority.HIGHEST);
+	}
+
+	// TODO: onWorldHopChange event in Plugin, update some manager/variable, and just display here
+	// 	     instead of getting world number every time from client. Not sure if this exists tho.
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		// Get orb location
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: This should have a config option for whether or not to show the overlay.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			Point worldOverlayLocation = widget.getCanvasLocation();
+			int x = worldOverlayLocation.getX();
+			int y = worldOverlayLocation.getY();
+//			int width = widget.getWidth();
+			int height = widget.getHeight();
+			int fontSize = 18;
+			int pixelSize = fontSize * 4 / 3;
+
+			Font currentFont = graphics.getFont();
+			graphics.setFont(new Font("Arial", Font.BOLD, fontSize));
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: I would like to see what this looks like using the normal RS fonts we use in the other overlays first.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	@Inject
+	private WorldHopperOverlay()
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		setPriority(OverlayPriority.HIGHEST);
+	}
+
+	// TODO: onWorldHopChange event in Plugin, update some manager/variable, and just display here
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: This is not something that should be done.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
+package net.runelite.client.plugins.worldhopper;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: New files should have a license header.

**Comment by Hydrox6** [comment]
Closing as stale. See linked issue

**Pull request closed without merging**